src/slang-utils/create-parser.ts

```diff
@@ -25,15 +32,48 @@ const milestoneVersions = Array.from(
     return versions;
   }, []);
 
+const bypassParse =
+  (parseOutput: NonterminalNode) => (): VersionExpressionSets => {
+    // We don't need to parse the text twice if we already have the
+    // NonterminalNode.
+    const parsed = new VersionExpressionSets(
+      new SlangVersionExpressionSets(parseOutput),
+      0
+    );
+    parsed.comments = [];
+    return parsed;
+  };
+
+const options = {
+  plugins: [
+    {
+      languages: [{ name: 'SolidityPragma', parsers: ['slangPragma'] }],
+      parsers: {
+        slangPragma: {
+          astFormat: 'slang-ast',
+          locStart,
+          locEnd
+        } as PrettierParser
+      },
+      printers: { ['slang-ast']: { print: slangPrint } }
+    }
+  ],
+  parser: 'slangPragma'
+};
+
+const query = Query.parse(
```

Comment:
I noticed you moved `query` out of the `tryToCollectPragmas` function, but you only use it there; should we move it back to its original location?

Reply:
this way the query is computed only once and not every time it tryToCollectPragmas is called